app/javascript/elm/tests/MobileSessionsFiltersTests.elm

```diff
@@ -15,6 +15,54 @@ import Test.Html.Selector as Slc
 import TimeRange
 
 
+locationFilter : Test
+locationFilter =
+    describe "Location filter tests:"
+        [ test "Location filter is present" <|
+            \_ ->
+                defaultModel
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.has [ Slc.id "location-filter" ]
+        , fuzz string "when user types UpdateLocationInput is triggered" <|
+            \locationValue ->
+                defaultModel
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.find [ Slc.id "location-filter" ]
+                    |> Event.simulate (Event.input locationValue)
+                    |> Event.expect (UpdateLocationInput locationValue)
+        , fuzz string "when UpdateLocationInput is triggered new input value is visible" <|
+            \locationValue ->
+                defaultModel
+                    |> update (UpdateLocationInput locationValue)
+                    |> Tuple.first
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.find [ Slc.id "location-filter" ]
+                    |> Query.has [ Slc.attribute <| Attr.value locationValue ]
+        , test "when Enter key is pressed SubmitLocation is triggered" <|
+            \_ ->
+                let
+                    simulatedEnterKeydown : Encode.Value
```

Comment:
Have you considered another name here? This is an encoded value, `simulated` seems not correct